shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -24,30 +24,44 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
     nme: Primitive,
     tparams: List[Primitive],
     body: Type,
+    mthDecls: List[MethodDef[Right[Term, Type]]],
+    mthDefs: List[MethodDef[Left[Term, Type]]],
     baseClasses: Set[Var],
     toLoc: Opt[Loc],
   ) {
     def allBaseClasses(ctx: Ctx)(implicit traversed: Set[Var]): Set[Var] =
       baseClasses.map(v => Var(v.name.toList.mapHead(_.toLower).mkString)) ++
         baseClasses.iterator.filterNot(traversed).flatMap(v =>
-          ctx.tyDefs(v.name).allBaseClasses(ctx)(traversed + v))
+          ctx.tyDefs.get(v.name).fold(Set.empty[Var])(_.allBaseClasses(ctx)(traversed + v)))
   }
   
-  case class Ctx(parent: Opt[Ctx], env: MutMap[Str, TypeScheme], lvl: Int, inPattern: Bool, tyDefs: Map[Str, TypeDef]) {
+  case class Ctx(parent: Opt[Ctx], env: MutMap[Str, TypeScheme], lvl: Int, inPattern: Bool, tyDefs: Map[Str, TypeDef],
+      mthDecls: Map[Str, Map[Str, PolymorphicType]], mthDefs: Map[Str, Map[Str, PolymorphicType]],
+      targsMaps: MutMap[Str, MutMap[Str, Map[Str, SimpleType]]], methodBase: MutMap[Str, Option[Var]]) {
     def +=(b: Binding): Unit = env += b
     def ++=(bs: IterableOnce[Binding]): Unit = bs.iterator.foreach(+=)
     def get(name: Str): Opt[TypeScheme] = env.get(name) orElse parent.dlof(_.get(name))(N)
     def contains(name: Str): Bool = env.contains(name) || parent.exists(_.contains(name))
-    def nest: Ctx = copy(Some(this), MutMap.empty)
+    def nest: Ctx = copy(Some(this), MutMap.empty, targsMaps = MutMap.empty)
     def nextLevel: Ctx = copy(lvl = lvl + 1)
+    def getTargsMaps(cls: Str): Option[MutMap[Str, Map[Str, SimpleType]]] =
+      targsMaps.get(cls) orElse parent.dlof(_.getTargsMaps(cls))(N)
+    def getTargsMaps(cls: Str, mth: Str): Option[Map[Str, SimpleType]] =
+      getTargsMaps(cls).dlof(_.get(mth) orElse parent.dlof(_.getTargsMaps(cls, mth))(N))(N)
+    def getTargsMaps(cls: Str, mth: Str, targ: Str): Option[SimpleType] =
```

Comment:
It seems this overload is not used.

Reply:
Updated direct references to `Ctx.targsMaps` to use this instead in 193deba9ab7b3358683405b91fdcb2f2f4dc1c54